packages/overlay/src/OverlayBase.ts

```diff
@@ -0,0 +1,908 @@
+            !this.triggerElement ||
+            !!(this.triggerElement as VirtualTrigger).updateBoundingClientRect
+        ) {
+            return;
+        }
+        this.bindEvents(this.triggerElement);
+    }
+
+    private elementIds: string[] = [];
+
+    private prepareLongpressDescription(trigger: HTMLElement): void {
+        if (
+            // only "longpress" relationships are described this way
+            this.triggerInteraction !== 'longpress' ||
+            // do not reapply until target it recycled
+            this.releaseLongpressDescribedby !== noop ||
+            // require "longpress content" to apply relationship
+            !this.elements.length
+        ) {
+            return;
+        }
+
+        const longpressDescription = document.createElement('div');
+        longpressDescription.id = `longpress-describedby-descriptor-${crypto
+            .randomUUID()
+            .slice(0, 8)}`;
+        const messageType = isIOS() || isAndroid() ? 'touch' : 'keyboard';
+        longpressDescription.textContent = LONGPRESS_INSTRUCTIONS[messageType];
+        longpressDescription.slot = 'longpress-describedby-descriptor';
+        trigger.insertAdjacentElement('afterend', longpressDescription);
+
+        const releaseLongpressDescribedby = conditionAttributeWithId(
+            trigger,
+            'aria-describedby',
+            [longpressDescription.id]
+        );
+        this.releaseLongpressDescribedby = () => {
+            releaseLongpressDescribedby();
+            longpressDescription.remove();
+            this.releaseLongpressDescribedby = noop;
+        };
+    }
+
+    private prepareAriaDescribedby(trigger: HTMLElement): void {
+        if (
+            // only "hover" relationships establed described by content
+            this.triggerInteraction !== 'hover' ||
+            // do not reapply until target is recycled
+            this.releaseAriaDescribedby !== noop ||
+            // require "hover content" to apply relationship
+            !this.elements.length
+        ) {
+            return;
+        }
+
+        const triggerRoot = trigger.getRootNode();
+        const contentRoot = this.elements[0].getRootNode();
+        const overlayRoot = this.getRootNode();
+        if (triggerRoot == overlayRoot) {
+            const releaseAriaDescribedby = conditionAttributeWithId(
+                trigger,
+                'aria-describedby',
+                [this.id]
+            );
+            this.releaseAriaDescribedby = () => {
+                releaseAriaDescribedby();
+                this.releaseAriaDescribedby = noop;
+            };
+        } else if (triggerRoot === contentRoot) {
+            this.elementIds = this.elements.map((el) => el.id);
+            const appliedIds = this.elements.map((el) => {
+                if (!el.id) {
+                    el.id = `${this.tagName.toLowerCase()}-helper-${crypto
+                        .randomUUID()
+                        .slice(0, 8)}`;
```

Comment:
Another `crypto.randomUUID()` here - ok as long as we never expect to execute this in HTTP (vs HTTPS)

Reply:
We'll cross that bridge if we ever get to it.